test/span.js

```diff
@@ -182,6 +186,147 @@ describe('span should', () => {
         assert.isOk(unnormalizedKey in Span._getBaggageHeaderCache());
     });
 
+    describe('adaptive sampling tests for span', () => {
+        let options = [
+            { desc: 'sampled: ', sampling: true, reportedSpans: 1 },
+            { desc: 'unsampled: ', sampling: false, reportedSpans: 0}
+        ];
+        _.each(options, (o) => {
+            it (o.desc + 'should save tags, and logs on an unsampled span incase it later becomes sampled', () => {
+                let reporter = new InMemoryReporter();
+                let tracer = new Tracer(
+                    'test-service-name',
+                    reporter,
+                    new ConstSampler(false),
+                    { logger: new MockLogger() }
+                );
+                let span = tracer.startSpan('initially-unsampled-span');
+                span.setTag('tagKeyOne', 'tagValueOne');
+                span.addTags({
+                    'tagKeyTwo': 'tagValueTwo'
+                });
+                span.log({'logkeyOne': 'logValueOne'});
+
+                tracer._sampler = new ConstSampler(o.sampling);
+                span.setOperationName('sampled-span');
+                span.finish();
+
+                assert.isOk(deepEqual(span._tags[0], {key: 'tagKeyOne', value: 'tagValueOne'}));
+                assert.isOk(deepEqual(span._tags[1], {key: 'tagKeyTwo', value: 'tagValueTwo'}));
+                assert.isOk(deepEqual(span._logs[0].fields[0], {key: 'logkeyOne', value: 'logValueOne'}));
+                assert.equal(reporter.spans.length, o.reportedSpans);
+            });
+        });
+
+        describe('span sampling finalizer', () => {
+            it ('should trigger when it inherits a sampling decision', () => {
+                // Span created in before each is not finalized.
```

Comment:
nits: 
  * usually methods like `assert.equal` take a message as the 3rd argument, so the text of the comment is better placed there, to serve both as documentation and a better error should the test fail.
  * s/before each/beforeEach()/